runtime/compiler/aarch64/codegen/J9MemoryReference.hpp

```diff
@@ -0,0 +1,134 @@
+
+/*
+ * The following #define and typedef must appear before any #includes in this file
+ */
+#ifndef J9_MEMORY_REFERENCE_CONNECTOR
+#define J9_MEMORY_REFERENCE_CONNECTOR
+
+namespace J9 { namespace ARM64 { class MemoryReference; } }
+namespace J9 { typedef J9::ARM64::MemoryReference MemoryReferenceConnector; }
+#else
+#error J9::ARM64::MemoryReference expected to be a primary connector, but a J9 connector is already defined
+#endif
+
+#include "codegen/OMRMemoryReference.hpp"
+
+namespace J9
+{
+
+namespace ARM64
+{
+
+class OMR_EXTENSIBLE MemoryReference : public OMR::MemoryReferenceConnector
+   {
+public:
+   /**
+    * @brief Constructor
+    * @param[in] cg : CodeGenerator object
+    */
+   MemoryReference(TR::CodeGenerator *cg)
+      : OMR::MemoryReferenceConnector(cg) {}
+
+   /**
+    * @brief Constructor
+    * @param[in] br : base register
+    * @param[in] ir : index register
+    * @param[in] cg : CodeGenerator object
+    */
+   MemoryReference(
+         TR::Register *br,
+         TR::Register *ir,
+         TR::CodeGenerator *cg)
+      : OMR::MemoryReferenceConnector(br, ir, cg) {}
+
+   /**
+    * @brief Constructor
+    * @param[in] br : base register
+    * @param[in] ir : index register
+    * @param[in] scale : scale of index
+    * @param[in] cg : CodeGenerator object
+    */
+   MemoryReference(
+         TR::Register *br,
+         TR::Register *ir,
+         uint8_t scale,
+         TR::CodeGenerator *cg)
+      : OMR::MemoryReferenceConnector(br, ir, scale, cg) {}
+
+   /**
+    * @brief Constructor
+    * @param[in] br : base register
+    * @param[in] disp : displacement
+    * @param[in] cg : CodeGenerator object
+    */
+   MemoryReference(
+         TR::Register *br,
+         int32_t disp,
+         TR::CodeGenerator *cg)
+      : OMR::MemoryReferenceConnector(br, disp, cg) {}
+
+   /**
+    * @brief Constructor
+    * @param[in] node : load or store node
+    * @param[in] len : length
+    * @param[in] cg : CodeGenerator object
+    */
+   MemoryReference(TR::Node *node, uint32_t len, TR::CodeGenerator *cg)
+      : OMR::MemoryReferenceConnector(node, len, cg) {}
+
+   /**
+    * @brief Constructor
+    * @param[in] node : node
+    * @param[in] symRef : symbol reference
+    * @param[in] len : length
+    * @param[in] cg : CodeGenerator object
+    */
+   MemoryReference(TR::Node *node, TR::SymbolReference *symRef, uint32_t len, TR::CodeGenerator *cg)
+      : OMR::MemoryReferenceConnector(node, symRef, len, cg) {}
+
+   /**
+    * @brief Estimates the length of generated binary
+    * @param[in] op : opcode of the instruction to attach this memory reference to
+    * @return estimated binary length
+    */
+   uint32_t estimateBinaryLength(TR::InstOpCode op);
+
+   /**
+    * @brief Generates binary encoding
+    * @param[in] ci : current instruction
+    * @param[in] cursor : instruction cursor
+    * @param[in] cg : CodeGenerator
+    * @return estimated binary length
```

Comment:
I think this is the advanced instruction cursor after the encoding.

Reply:
Corrected the comment.  I brought the comment from `OMRMemoryReference.hpp`, and I am going to fix it separately.